netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -1699,6 +1699,14 @@ net_ebpf_extension_sock_addr_redirect_connection_classify(
         _net_ebpf_ext_insert_connection_context_to_list(blocked_connection_context);
 
         InterlockedIncrement(&_net_ebpf_ext_statistics.block_connection_count);
+    } else {
+        // Remove any 'stale' connection context if found.
```

Comment:
nit: maybe we can add some more details on why it is possible to find a stale connection context?